lib/truffle/truffle/cext.rb

```diff
@@ -1801,20 +1770,14 @@ def rb_thread_call_without_gvl(function, data1, unblock, data2)
     Thread.current.unblock unblocker, runner
   end
 
-  def rb_iterate_call_block( iter_block, block_arg, arg2, &block)
-    execute_with_mutex iter_block, block_arg, arg2, &block
-  end
-
-  def rb_iterate(function, arg1, iter_block, arg2, block)
-    if block
-      call_c_with_block function, arg1 do |block_arg|
-        rb_iterate_call_block(iter_block, block_arg, arg2) do |*args|
-          block.call(*args)
-        end
+  def rb_iterate(iteration, iterated_object, callback, callback_arg)
+    if (block = rb_block_proc)
+      call_with_thread_locally_stored_block iteration, iterated_object do |block_arg|
+        rb_iterate_call_block(callback, block_arg, callback_arg) { |*args| block.call(*args) }
```

Comment:
Can the block be passed like `&block`?

Reply:
where?

Reply:
Instead of `) { |*args| block.call(*args) }`

Reply:
done